implementation/src/main/java/io/smallrye/config/ConfigValueMapView.java

```diff
@@ -0,0 +1,127 @@
+package io.smallrye.config;
+
+import java.util.AbstractCollection;
+import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The ConfigValueMapView is view over a Map of String configs names and ConfigValue value.
+ * <p>
+ *
+ * Use this to wrap the ConfigValue map and expose it where a Map of String name and String value is required.
+ */
+public class ConfigValueMapView extends AbstractMap<String, String> {
+    private final Map<String, ConfigValue> delegate;
+
+    ConfigValueMapView(final Map<String, ConfigValue> delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public int size() {
+        return delegate.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return delegate.isEmpty();
+    }
+
+    @Override
+    public boolean containsKey(final Object key) {
+        return delegate.containsKey(key);
+    }
+
+    @Override
+    public boolean containsValue(final Object value) {
+        return values().contains(value);
+    }
+
+    @Override
+    public String get(final Object key) {
+        final ConfigValue configValue = delegate.get(key);
+        return configValue != null ? configValue.getValue() : null;
+    }
+
+    private transient Set<String> keySet;
+    private transient Set<Map.Entry<String, String>> entrySet;
+    private transient Collection<String> values;
+
+    @Override
+    public Set<String> keySet() {
+        if (keySet == null) {
+            keySet = Collections.unmodifiableSet(delegate.keySet());
+        }
+        return keySet;
+    }
+
+    @Override
+    public Set<Map.Entry<String, String>> entrySet() {
+        if (entrySet == null) {
+            entrySet = Collections.unmodifiableSet(new AbstractSet<Entry<String, String>>() {
```

Comment:
If you are using `unmodifiableMap` above, you don't need `unmodifiableSet` here (it does it for you already).